src/lib.rs

```diff
@@ -1140,6 +1140,20 @@ impl<'a> Arbitrary<'a> for Ipv6Addr {
     }
 }
 
+impl<'a> Arbitrary<'a> for IpAddr {
+    fn arbitrary(u: &mut Unstructured<'a>) -> Result<Self> {
+        if u.arbitrary()? {
+            u.arbitrary().map(IpAddr::V4)
+        } else {
+            u.arbitrary().map(IpAddr::V6)
+        }
```

Comment:
I find it much easier to read this code when written like this:

```suggestion
        if u.arbitrary()? {
            Ok(IpAddr::V4(u.arbitrary()?))
        } else {
            Ok(IpAddr::V6(u.arbitrary()?))
        }
```